diff -Nur ntp-4.2.8p12.bak/ntpd/ntp_crypto.c ntp-4.2.8p12/ntpd/ntp_crypto.c
--- ntp-4.2.8p12.bak/ntpd/ntp_crypto.c	2019-01-22 20:41:39.485000000 -0500
+++ ntp-4.2.8p12/ntpd/ntp_crypto.c	2019-01-22 20:42:07.748000000 -0500
@@ -3387,9 +3387,6 @@
 		 */
 		peer->crypto |= CRYPTO_FLAG_CERT;
 		peer->grpkey = yp->grpkey;
-		if (peer->ident == NULL || !(peer->crypto &
-		    CRYPTO_FLAG_MASK))
-			peer->crypto |= CRYPTO_FLAG_VRFY;
 	}
 
 	/*
