From 302cec597f3ab4e522dacc3563953d10c0f7e656 Mon Sep 17 00:00:00 2001
From: zhangyao <zhangyao65@huawei.com>
Date: Thu, 25 Jul 2019 16:42:51 +0000
Subject: [PATCH] revert-ntpd-fix-autokey-condition-error


---
 ntpd/ntp_crypto.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ntpd/ntp_crypto.c b/ntpd/ntp_crypto.c
index 53f3253..62ea72c 100644
--- a/ntpd/ntp_crypto.c
+++ b/ntpd/ntp_crypto.c
@@ -3387,6 +3387,9 @@ cert_hike(
 		 */
 		peer->crypto |= CRYPTO_FLAG_CERT;
 		peer->grpkey = yp->grpkey;
+                if (peer->ident == NULL || !(peer->crypto &
+                    CRYPTO_FLAG_MASK))
+                        peer->crypto |= CRYPTO_FLAG_VRFY;
 	}
 
 	/*
-- 
1.8.3.1

